### PR TITLE
Move RazorViewEngineOptionsSetup to internal namespace

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #6116 

cc @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs [hunk 1]
```diff
@@ -4,11 +4,10 @@
 using System;
 using System.Linq;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.AspNetCore.Mvc.Razor;
 using Microsoft.CodeAnalysis;
 using Microsoft.Extensions.Options;
 
-namespace Microsoft.AspNetCore.Mvc
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs]: Remove the `Obsolete` from the type now 👍

**Comment by jbagga** [comment]
@rynowak Updated

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs [hunk 2]
```diff
@@ -4,16 +4,14 @@
...
 {
     /// <summary>
     /// Sets up default options for <see cref="RazorViewEngineOptions"/>.
     /// </summary>
-    [Obsolete("This type is for internal use and will be removed in a future version.")]
     public class RazorViewEngineOptionsSetup : ConfigureOptions<RazorViewEngineOptions>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs]: Could you also change this type to derive from `IConfigureOptions<RazorViewEngineOptions>`. Look at the other OptionsSetup type to see what they do.

**Comment by jbagga** [comment]
@pranavkm Updated

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs [hunk 3]
```diff
@@ -12,51 +12,55 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
...
+            _hostingEnvironment = hostingEnvironment;
         }
 
-        private static void ConfigureRazor(
-            RazorViewEngineOptions razorOptions,
-            IHostingEnvironment hostingEnvironment)
+        public void Configure(RazorViewEngineOptions options)
         {
-            if (hostingEnvironment.ContentRootFileProvider != null)
+            if (_hostingEnvironment.ContentRootFileProvider != null)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs]:
```
if (options == null)
{
   throw new ArgumentNullException(nameof(options));
}
```

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**